komodo/build.py

```diff
@@ -77,58 +79,72 @@ def cmake(pkg, ver, path, prefix, builddir,
         print(shell('make -j{}'.format(jobs)))
         print(shell('make DESTDIR={} install'.format(fakeroot)))
 
-def sh(pkg, ver, pkgpath, prefix, makefile, *args, **kwargs):
-    if os.path.isfile(makefile):
-        makefile = os.path.abspath(makefile)
+
+def sh(pkg, ver, pkgpath, data, prefix, makefile, *args, **kwargs):
+    makefile = data.get(makefile)
 
     with pushd(pkgpath):
-        makefile = os.path.abspath(makefile)
         cmd = ['bash {} --prefix {}'.format(makefile, prefix),
                '--fakeroot {}'.format(kwargs['fakeroot']),
                '--python {}/bin/python'.format(prefix)]
         if 'jobs' in kwargs:
             cmd.append('--jobs {}'.format(kwargs['jobs']))
         if 'cmake' in kwargs:
             cmd.append('--cmake {}'.format(kwargs['cmake']))
+        cmd.append('--pythonpath {}'.format(kwargs['pythonpath']))
+        cmd.append('--path {}'.format(kwargs['binpath']))
+        cmd.append('--pip {}'.format(kwargs['pip']))
+        cmd.append('--virtualenv {}'.format(kwargs['virtualenv']))
         cmd.append(kwargs.get('makeopts'))
 
         print('Installing {} ({}) from sh'.format(pkg, ver))
         shell(cmd)
 
-def rsync(pkg, ver, pkgpath, prefix, *args, **kwargs):
+
+def rsync(pkg, ver, pkgpath, data, prefix, *args, **kwargs):
     print('Installing {} ({}) with rsync'.format(pkg, ver))
     # assume a root-like layout in the pkgpath dir, and just copy it
     shell(['rsync -am', kwargs.get('makeopts'), '{}/'.format(pkgpath), kwargs['fakeroot'] + prefix])
 
 
-def pip_install(pkg, ver, pkgpath, prefix, dlprefix, pip='pip', *args, **kwargs):
+def pip_install(pkg, ver, pkgpath, data, prefix, dlprefix, pip='pip', *args, **kwargs):
     cmd = [pip,
            'install {}=={}'.format(pkg, komodo.strip_version(ver)),
            '--root {}'.format(kwargs['fakeroot']),
            '--prefix {}'.format(prefix),
-           '--force-reinstall',
            '--no-index',
            '--no-deps',
+           '--ignore-installed',
            '--find-links {}'.format(dlprefix),
-           kwargs.get('makeopts', '')]
+    ]
+    cmd.append(kwargs.get('makeopts', ''))
```

Comment:
Why use append instead of just having this as the last list elem as in the previous version?

Reply:
Yeah, there was something in between there that has since been removed. Good catch